src/compute/src/render/top_k.rs

```diff
@@ -356,23 +356,19 @@ where
     where
         S: Scope<Timestamp = G::Timestamp>,
     {
-        let pairer = Pairer::new(1);
         let input = collection.map(move |(hash_key, row)| {
             let mut hash_key_iter = hash_key.iter();
             let hash = hash_key_iter.next().unwrap().unwrap_uint64() % modulus;
-            let hash_key = pairer.merge(std::iter::once(Datum::from(hash)), hash_key_iter);
+            let hash_key = SharedRow::pack(std::iter::once(hash.into()).chain(hash_key_iter));
             (hash_key, row)
         });
-        let (oks, errs) = if validating {
-            let stage = build_topk_negated_stage::<S, _, _, RowValSpine<Result<Row, Row>, _, _>>(
-                &input,
-                |v| v.into_owned(),
-                order_key,
-                offset,
-                limit,
-                arity,
-            )
-            .as_collection(|k, v| (SharedRow::pack(k), v.clone()));
+        let (input, oks, errs) = if validating {
+            let from = |v: &Result<Row, Row>| v.into_owned();
+            let (input, stage) =
+                build_topk_negated_stage::<S, _, _, RowValSpine<Result<Row, Row>, _, _>>(
+                    &input, from, order_key, offset, limit, arity,
+                );
+            let stage = stage.as_collection(|k, v| (SharedRow::pack(k), v.clone()));
```

Comment:
Is it right that the `from` movement is tidying, and the only thing going on here is returning the other result returned from `build_topk_negated_stage`?

Reply:
Yes, this is to avoid the horrors of `cargo fmt`, which would spread it over approximately 150 lines otherwise.